artifactory/services/remoteRepository.go

```diff
@@ -0,0 +1,353 @@
+	Username                          string                 `json:"username,omitempty"`
+	Password                          string                 `json:"password,omitempty"`
+	Proxy                             string                 `json:"proxy,omitempty"`
+	Description                       string                 `json:"description,omitempty"`
+	Notes                             string                 `json:"notes,omitempty"`
+	IncludesPattern                   string                 `json:"includesPattern,omitempty"`
+	ExcludesPattern                   string                 `json:"excludesPattern,omitempty"`
+	RepoLayoutRef                     string                 `json:"repoLayoutRef, omitempty"`
+	HardFail                          bool                   `json:"hardFail, omitempty"`
+	Offline                           bool                   `json:"offline, omitempty"`
+	BlackedOut                        bool                   `json:"blackedOut, omitempty"`
+	StoreArtifactsLocally             bool                   `json:"storeArtifactsLocally, omitempty"`
+	SocketTimeoutMillis               int                    `json:"socketTimeoutMillis, omitempty"`
+	LocalAddress                      string                 `json:"localAddress, omitempty"`
+	RetrievalCachePeriodSecs          int                    `json:"retrievalCachePeriodSecs, omitempty"`
+	FailedRetrievalCachePeriodSecs    int                    `json:"failedRetrievalCachePeriodSecs, omitempty"`
+	MissedRetrievalCachePeriodSecs    int                    `json:"missedRetrievalCachePeriodSecs, omitempty"`
+	UnusedArtifactsCleanupEnabled     bool                   `json:"unusedArtifactsCleanupEnabled, omitempty"`
+	UnusedArtifactsCleanupPeriodHours int                    `json:"unusedArtifactsCleanupPeriodHours, omitempty"`
+	AssumedOfflinePeriodSecs          int                    `json:"assumedOfflinePeriodSecs, omitempty"`
+	ShareConfiguration                bool                   `json:"shareConfiguration, omitempty"`
+	SynchronizeProperties             bool                   `json:"synchronizeProperties, omitempty"`
+	BlockMismatchingMimeTypes         bool                   `json:"blockMismatchingMimeTypes, omitempty"`
+	PropertySets                      []string               `json:"propertySets, omitempty"`
+	AllowAnyHostAuth                  bool                   `json:""allowAnyHostAuth", omitempty"`
+	EnableCookieManagement            bool                   `json:"enableCookieManagement, omitempty"`
+	BypassHeadRequests                bool                   `json:"bypassHeadRequests, omitempty"`
+	ClientTlsCertificate              string                 `json:"clientTlsCertificate, omitempty"`
+	BlockPushingSchema1               bool                   `json:"blockPushingSchema1, omitempty"`
+	contentSynchronisation            ContentSynchronisation `json:"contentSynchronisation, omitempty"`
+}
+
+type MavenGradleRemoteRepositoryParams struct {
```

Comment:
I suggest splitting this struct into two structs - MavenRemoteRepositoryParams and GradleRemoteRepositoryParams. We can put the common fields of the two structs in a third struct.
This also applies to LocalRepositoryService and VirtualRepositoryService.